daemon/cluster/executor/container/controller.go

```diff
@@ -153,20 +154,39 @@ func (r *controller) Wait(pctx context.Context) error {
 	ctx, cancel := context.WithCancel(pctx)
 	defer cancel()
 
+	healthErr := make(chan error, 1)
+	go func() {
+		ectx, cancel := context.WithCancel(ctx) // cancel event context on first event
+		defer cancel()
+		if err := r.checkHealth(ectx); err == ErrContainerUnhealthy {
+			healthErr <- ErrContainerUnhealthy
+			if err := r.Shutdown(ectx); err != nil {
+				log.G(ectx).WithError(err).Debug("shutdown failed on unhealthy")
+			}
+		}
+	}()
+
 	err := r.adapter.wait(ctx)
```

Comment:
I am having issues with an image using health checks, it uses curl localhost:port and from my logs at the swarm service inspect command the error I got is from the swarm node hosting the container being checked.
I supposed more like a docker exec way inside the container to be checked.
Thanks for any feedback.